examples/type_inference/type_inference_visitor_unittest.py

```diff
@@ -0,0 +1,243 @@
+    """testers specifically for the function set_const using single nodes."""
+
+    def test_const_str(self):
+        """testing if the function set_const has the correct output for node
+        type str."""
+        expected = str
+        node = test_utils.extract_node("""a = __('sample_string')""")
+        set_const(node)
+        self.assertEqual(expected, node.type_constraints)
+
+    def test_const_int(self):
+        """testing if the function set_const has the correct output for node
+        type int."""
+        expected = int
+        node = test_utils.extract_node("""a = __(5)""")
+        set_const(node)
+        self.assertEqual(expected, node.type_constraints)
+
+    def test_const_float(self):
+        """testing if the function set_const has the correct output for node
+        type float."""
+        expected = float
+        node = test_utils.extract_node("""a = __(2.18)""")
+        set_const(node)
+        self.assertEqual(expected, node.type_constraints)
+
+    def test_const_bool(self):
+        """testing if the function set_const has the correct output for node
+        type boolean."""
+        expected = bool
+        node = test_utils.extract_node("""a = __(True)""")
+        set_const(node)
+        self.assertEqual(expected, node.type_constraints)
+
+    def test_const_none(self):
+        """testing if the function set_const has the correct output for
+        NoneType node."""
+        expected = type(None)
+        node = test_utils.extract_node("""a = __(None)""")
+        set_const(node)
+        self.assertEqual(expected, node.type_constraints)
+
+
+class TypeInferenceVisitorTest(unittest.TestCase):
+    """testers for type_inference_visitor. Modules are been passed in instead
+    of single nodes."""
+
+    def test_binop_same_type_operands(self):
+        """testing if a binary operator passed into TypeVisitor
+        has the correct type_constraints attribute when both operands have
+        the same type.
+
+        Note: astroid.parse() and type_visitor both have a return type
+        Module(), if we directly look for nodes under Module(), the node
+        binop will not be returned, but instead, it will be a wrapper for
+        binop -> astroid.Expr. To find the binop wrapped by the Expr node,
+        we need to use Expr.value. So binop is 2 layers under the return
+        type of astroid.parse and type_visitor.visit (binop is 2 layers
+        under module).
+
+        An example from AST:
+        > parseprint('-a')
+        Module(body=[
+        Expr(value=UnaryOp(op=USub(), operand=Name(id='a', ctx=Load()))),
+        ])
+        """
+        type_visitor = TypeVisitor()
+        type_visitor.register_transform(astroid.Const, set_const)
+        type_visitor.register_transform(astroid.BinOp, set_binop)
+        module = astroid.parse("""10 + 2""")    # int
+        transformed_module = type_visitor.visit(module)
+        for node in transformed_module.body:
```

Comment:
I changed the code to 
`visited_module = type_visitor.visit(module)
self.assertEqual(Dict, visited_module.body[0].value.type_constraints)` instead of the for loop. It gives desired result. But I'm not sure what `nodes_of_class` is for.. since I'm checking the type constraints, I called `node.type_constarints` and comparing it with the correct class type.